src/main/java/mekanism/common/tile/prefab/TileEntityMultiblock.java

```diff
@@ -232,9 +218,6 @@ public void setRemoved() {
         super.setRemoved();
         if (!isRemote()) {
             structure.invalidate(level);
-            if (cachedID != null) {
```

Comment:
is setRemoved handled elsewhere?

Reply:
It doesn't need to be because of things getting new ids that if they point to a cache that ends up not existing at some point then we can just treat them as not having a cache.

Reply:
Perhaps this is my ignorance, when is this method called? (I've not had a chance to open this in an ide)

Reply:
`setRemoved` is called when the tile is removed (broken) or the chunk it is in unloads. As the MultiblockManager no longer keeps track of a id -> cache + positions map and only keeps track of an id -> cache map there is no need to be removing the position from the manager (which is what the invalidate method was doing)

Reply:
OK, my assumption was correct there.

So it doesn't need to mark the multiblock as invalid? (because part of it was removed/unloaded)

Reply:
That is what the `structure.invalidate(level)` line is doing. The `MultiblockManager#invalidate` was purely for invalidating it in regards to the cache.

Reply:
Cool, thanks for explaining!